Add references for four methods in the code not described in the book (#1216)
browserengineering/book: book/scheduling.md

        self.active_tab.task_runner.clear_pending_tasks()
        task = Task(self.active_tab.load, url, body)
        self.active_tab.task_runner.schedule_task(task)
```

Above, I needed to clear any pending tasks before loading a new page, because
those previous tasks are now invalid:

``` {.python}
class TaskRunner:
    def clear_pending_tasks(self):
        self.condition.acquire(blocking=True)
        self.tasks.clear()
        self.pending_scroll = None
        self.condition.release()
```

We also need to split `new_tab` into a version that acquires a lock
and one that doesn't (`new_tab_internal`):

``` {.python}

>     def new_tab(self, url):

Reviewer: ```suggestion
class Browser:
    def new_tab(self, url):
```

        self.lock.acquire(blocking=True)
        self.new_tab_internal(url)
        self.lock.release()

    def new_tab_internal(self, url):
        new_tab = Tab(self, HEIGHT - self.chrome.bottom)
        self.tabs.append(new_tab)
        self.set_active_tab(new_tab)
        self.schedule_load(url)
```

> ...because cases like the `click` method on `Chrome` are called by
> `handle_click`, which has already acquired the lock.^[Using locks properly and
> avoiding race conditions and deadlocks can be quite difficult!]

Reviewer: ```suggestion
This way `new_tab_internal` can be called directly by methods,
like `Chrome`'s `click` method, that already hold the lock.^[Using locks while
avoiding race conditions and deadlocks can be quite difficult!]
```

``` {.python}
class Chrome:
    def click(self, x, y):

>         # ...
>             self.browser.new_tab_internal(
>                 URL("https://browser.engineering/"))

Reviewer: ```suggestion
        if self.newtab_rect.contains(x, y):
            self.browser.new_tab_internal(
                URL("https://browser.engineering/"))
```

I believe we never indent without including the line that indents.

``` {.python}
class Browser:
    def handle_click(self, e):

>         self.lock.acquire(blocking=True)

Reviewer: A bug in the chapter?

Author: yes
